Create rmd.yml

tschm opened the pull request:
Convert all notebooks into rmd files and publish them on a separate branch. Do not remove the rmd files from the main branch yet.

tschm commented:
The generated rmd files can be seen here: https://github.com/tschm/ISLP_labs/tree/rmd-files

jonathan-taylor commented:
So, the idea is to run this action on a commit or tag and have these automatically generated? That seems fine, then we can leave a link in the README.md to this branch?

tschm commented:
There is also a new release job. Go to

Actions => Bump version and publish (on left). 

It will introduce a new tag and a release.

jonathan-taylor commented:
Yes, I saw that.

Thanks for this work on the actions! Haven't tried them yet, but this is great.
________________________________
From: Thomas Schmelzer ***@***.***>
Sent: Wednesday, April 9, 2025 8:47 PM
To: intro-stat-learning/ISLP_labs ***@***.***>
Cc: Jonathan Taylor ***@***.***>; Comment ***@***.***>
Subject: Re: [intro-stat-learning/ISLP_labs] Create rmd.yml (PR #50)


There is also a new release job. Go to

Actions => Bump version and publish (on left).

It will introduce a new tag and a release.

—
Reply to this email directly, view it on GitHub<https://github.com/intro-stat-learning/ISLP_labs/pull/50#issuecomment-2791476467>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AACTM234SCFYTGM32KY6FU32YXSW7AVCNFSM6AAAAAB2NYQBYWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOOJRGQ3TMNBWG4>.
You are receiving this because you commented.Message ID: ***@***.***>

[https://avatars.githubusercontent.com/u/2046079?s=20&v=4]tschm left a comment (intro-stat-learning/ISLP_labs#50)<https://github.com/intro-stat-learning/ISLP_labs/pull/50#issuecomment-2791476467>

There is also a new release job. Go to

Actions => Bump version and publish (on left).

It will introduce a new tag and a release.

—
Reply to this email directly, view it on GitHub<https://github.com/intro-stat-learning/ISLP_labs/pull/50#issuecomment-2791476467>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AACTM234SCFYTGM32KY6FU32YXSW7AVCNFSM6AAAAAB2NYQBYWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOOJRGQ3TMNBWG4>.
You are receiving this because you commented.Message ID: ***@***.***>

tschm commented:
@jonathan-taylor  So, I have never worked with Rmd files. I assume you can run them using RStudio? However, the files (e.g. auto.csv, etc.) are not copied with them to the new branch. I think it would be good if we put all files needed by the notebooks into a folder data and copy this folder also for the “rmd-files” branch?

jonathan-taylor commented:
Yes, this makes sense. There are a handful of files needed for the notebooks to run to completion.

At least


  *
Auto.{csv,data}
  *
book_images/*
  *
imagenet_class_index.json

Yes, they can be opened in RStudio. I also find it handy to have a text notebook around as it's easier to edit and read than .ipynb.
________________________________
From: Thomas Schmelzer ***@***.***>
Sent: Thursday, April 10, 2025 9:07 PM
To: intro-stat-learning/ISLP_labs ***@***.***>
Cc: Jonathan Taylor ***@***.***>; Mention ***@***.***>
Subject: Re: [intro-stat-learning/ISLP_labs] Create rmd.yml (PR #50)


@jonathan-taylor<https://github.com/jonathan-taylor> So, I have never worked with Rmd files. I assume you can run them using RStudio? However, the files (e.g. auto.csv, etc.) are not copied with them to the new branch. I think it would be good if we put all files needed by the notebooks into a folder data and copy this folder also for the “rmd-files” branch?

—
Reply to this email directly, view it on GitHub<https://github.com/intro-stat-learning/ISLP_labs/pull/50#issuecomment-2795783340>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AACTM26KNDLT7E4Q2FEM64L2Y45YZAVCNFSM6AAAAAB2NYQBYWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOOJVG44DGMZUGA>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

[https://avatars.githubusercontent.com/u/2046079?s=20&v=4]tschm left a comment (intro-stat-learning/ISLP_labs#50)<https://github.com/intro-stat-learning/ISLP_labs/pull/50#issuecomment-2795783340>

@jonathan-taylor<https://github.com/jonathan-taylor> So, I have never worked with Rmd files. I assume you can run them using RStudio? However, the files (e.g. auto.csv, etc.) are not copied with them to the new branch. I think it would be good if we put all files needed by the notebooks into a folder data and copy this folder also for the “rmd-files” branch?

—
Reply to this email directly, view it on GitHub<https://github.com/intro-stat-learning/ISLP_labs/pull/50#issuecomment-2795783340>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AACTM26KNDLT7E4Q2FEM64L2Y45YZAVCNFSM6AAAAAB2NYQBYWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOOJVG44DGMZUGA>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

tschm commented:
@jonathan-taylor I am introducing a “data” folder and copy it over during ci/cd. Currently the tests are running.

tschm commented:
I have added an example how I think about using the ipynb files as templates. I have added the file ChXX (you can remove it later) with {{ author }} somewhere in a Markdown cell. I then have two different config files. One for rmd files and one for LaTeX files. In render.py I use Jinja to inject the desired values. Please check the rmd files branch in my fork of your repo to see the results... Please try to merge #61 and #65
